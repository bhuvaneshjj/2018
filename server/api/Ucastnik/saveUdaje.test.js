'use strict';

const db = require('../../db');
const Actions = require('../../../common/common');
const createWsServer = require('../../createWsServer');
const createWsClient = require('./../createWsClient');
const Ucastnik = require('../../model/Ucastnik/Ucastnik');
const generateTestToken = require('../generateTestToken');

const port = 5601;
const wsServer = createWsServer({});
const wsClient = createWsClient({ port });

beforeAll(async () => {
  wsServer.httpServer().listen(port);
  await wsClient.open();

  await db.dropDatabase();
});

afterAll(async () => {
  await wsClient.close();
  wsServer.httpServer().close();

  await db.disconnect();
});

it('vytvoř minimálního účastníka', async () => {
  const udaje = {
    prijmeni: 'Balabák',
    jmeno: 'František',
    narozeni: { rok: 1953 },
    pohlavi: 'muž',
    obec: 'Ostrava 1'
  };

  const { requestId, ...response } = await wsClient.sendRequest(
    Actions.saveUdaje({ rok: 2018, udaje }, generateTestToken())
  );
  expect(response.response.id).not.toBeNull();
  response.response.id = '---';
  expect(response).toMatchSnapshot();

  const ucastnici = await Ucastnik.find({}, { _id: 0, __v: 0 });
  expect(ucastnici).toMatchSnapshot();

  await Ucastnik.collection.drop();
});

it('vytvoř dvě účasti', async () => {
  const udaje1 = {
    prijmeni: 'Sukdoláková',
    jmeno: 'Božena',
    narozeni: { rok: 1967 },
    pohlavi: 'žena',
    obec: 'Kladno Rozdělov',
    psc: '327 41'
  };
  const udaje2 = { ...udaje1, obec: 'Kamenický Přívoz' };

  const response1 = await wsClient.sendRequest(
    Actions.saveUdaje({ rok: 2017, udaje: udaje1 }, generateTestToken())
  );

  const ucastnikId = response1.response.id;
  const { requestId, ...response2 } = await wsClient.sendRequest(
    Actions.saveUdaje({ id: ucastnikId, rok: 2018, udaje: udaje2 }, generateTestToken())
  );
  response2.response.id = '---';
  expect(response2).toMatchSnapshot();

  const ucastnici = await Ucastnik.find({}, { _id: 0, __v: 0 });
  expect(ucastnici).toMatchSnapshot();

  await Ucastnik.collection.drop();
});

it('přepiš existující účast', async () => {
  const udaje1 = {
    prijmeni: 'Sukdoláková',
    jmeno: 'Božena',
    narozeni: { rok: 1967 },
    pohlavi: 'žena',
    obec: 'Kladno 1',
    psc: '327 41'
  };
  const udaje2 = { ...udaje1, obec: 'Kladno 2' };
  const udaje3 = { ...udaje1, obec: 'Kladno 3' };
  const udaje4 = { ...udaje1, obec: 'Kladno 4' };
  const udaje5 = { ...udaje1, obec: 'Kladno 5' };

  const response1 = await wsClient.sendRequest(
    Actions.saveUdaje({ rok: 2018, udaje: udaje1 }, generateTestToken())
  );
  const ucastnikId = response1.response.id;

  await wsClient.sendRequest(
    Actions.saveUdaje({ id: ucastnikId, rok: 2017, udaje: udaje2 }, generateTestToken())
  );
  await wsClient.sendRequest(
    Actions.saveUdaje({ id: ucastnikId, rok: 2016, udaje: udaje3 }, generateTestToken())
  );
  await wsClient.sendRequest(
    Actions.saveUdaje({ id: ucastnikId, rok: 2015, udaje: udaje4 }, generateTestToken())
  );

  const { requestId, ...response5 } = await wsClient.sendRequest(
    Actions.saveUdaje({ id: ucastnikId, rok: 2017, udaje: udaje5 }, generateTestToken())
  );
  response5.response.id = '---';
  expect(response5).toMatchSnapshot();

  const ucastnici = await Ucastnik.find({}, { _id: 0, __v: 0 });
  expect(ucastnici).toMatchSnapshot();

  await Ucastnik.collection.drop();
});

it('saveUcast [not authenticated]', async () => {
  const { requestId, ...response } = await wsClient.sendRequest(Actions.saveUdaje({}, null));
  expect(response).toMatchSnapshot();
});
