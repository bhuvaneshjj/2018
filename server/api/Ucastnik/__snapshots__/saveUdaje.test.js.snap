// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`přepiš existující účast 1`] = `
Object {
  "code": "ok",
  "response": Object {
    "id": "---",
  },
  "status": "uloženo v pořádku",
}
`;

exports[`přepiš existující účast 2`] = `
Array [
  Object {
    "ucasti": Array [
      Object {
        "platby": Array [],
        "rok": 2018,
        "ubytovani": Array [],
        "udaje": Object {
          "jmeno": "Božena",
          "narozeni": Object {
            "rok": 1967,
          },
          "obec": "Kladno 1",
          "pohlavi": "žena",
          "prijmeni": "Sukdoláková",
          "psc": "327 41",
          "stat": "Česká republika",
        },
      },
      Object {
        "platby": Array [],
        "rok": 2017,
        "ubytovani": Array [],
        "udaje": Object {
          "jmeno": "Božena",
          "narozeni": Object {
            "rok": 1967,
          },
          "obec": "Kladno 5",
          "pohlavi": "žena",
          "prijmeni": "Sukdoláková",
          "psc": "327 41",
          "stat": "Česká republika",
        },
      },
      Object {
        "platby": Array [],
        "rok": 2016,
        "ubytovani": Array [],
        "udaje": Object {
          "jmeno": "Božena",
          "narozeni": Object {
            "rok": 1967,
          },
          "obec": "Kladno 3",
          "pohlavi": "žena",
          "prijmeni": "Sukdoláková",
          "psc": "327 41",
          "stat": "Česká republika",
        },
      },
      Object {
        "platby": Array [],
        "rok": 2015,
        "ubytovani": Array [],
        "udaje": Object {
          "jmeno": "Božena",
          "narozeni": Object {
            "rok": 1967,
          },
          "obec": "Kladno 4",
          "pohlavi": "žena",
          "prijmeni": "Sukdoláková",
          "psc": "327 41",
          "stat": "Česká republika",
        },
      },
    ],
  },
]
`;

exports[`saveUcast [not authenticated] 1`] = `
Object {
  "code": "authentication token invalid",
  "status": "Špatný ověřovací token. Zkus se přihlásit znovu. Detaily: JsonWebTokenError: jwt must be provided",
}
`;

exports[`vytvoř dvě účasti 1`] = `
Object {
  "code": "ok",
  "response": Object {
    "id": "---",
  },
  "status": "uloženo v pořádku",
}
`;

exports[`vytvoř dvě účasti 2`] = `
Array [
  Object {
    "ucasti": Array [
      Object {
        "platby": Array [],
        "rok": 2017,
        "ubytovani": Array [],
        "udaje": Object {
          "jmeno": "Božena",
          "narozeni": Object {
            "rok": 1967,
          },
          "obec": "Kladno Rozdělov",
          "pohlavi": "žena",
          "prijmeni": "Sukdoláková",
          "psc": "327 41",
          "stat": "Česká republika",
        },
      },
      Object {
        "platby": Array [],
        "rok": 2018,
        "ubytovani": Array [],
        "udaje": Object {
          "jmeno": "Božena",
          "narozeni": Object {
            "rok": 1967,
          },
          "obec": "Kamenický Přívoz",
          "pohlavi": "žena",
          "prijmeni": "Sukdoláková",
          "psc": "327 41",
          "stat": "Česká republika",
        },
      },
    ],
  },
]
`;

exports[`vytvoř minimálního účastníka 1`] = `
Object {
  "code": "ok",
  "response": Object {
    "id": "---",
  },
  "status": "uloženo v pořádku",
}
`;

exports[`vytvoř minimálního účastníka 2`] = `
Array [
  Object {
    "ucasti": Array [
      Object {
        "platby": Array [],
        "rok": 2018,
        "ubytovani": Array [],
        "udaje": Object {
          "jmeno": "František",
          "narozeni": Object {
            "rok": 1953,
          },
          "obec": "Ostrava 1",
          "pohlavi": "muž",
          "prijmeni": "Balabák",
          "stat": "Česká republika",
        },
      },
    ],
  },
]
`;
